src/arrayMethodJoin.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinString = '';
+    let stringSeparator = '';
+
+    separator === undefined
+      ? stringSeparator = ','
+      : stringSeparator = '' + separator;
```

Comment:
set default separator while describing function arguments instead of using separate condition